indico/modules/categories/controllers/display.py

```diff
@@ -347,3 +348,158 @@ def _process(self):
+            template = 'display/overview/week.html'
+            params['previous_week_url'] = self._other_day_url(self.start_dt - relativedelta(days=7))
+            params['next_week_url'] = self._other_day_url(self.start_dt + relativedelta(days=7))
+        elif self.period == 'month':
+            days = self._get_calendar_days()
+            template = 'display/overview/month.html'
+
+        events_by_day = []
+        for day in days:
+            events_by_day.append((day, self._pop_head_while(lambda x: x.start_dt.date() <= day.date(), events)))
+
+        # Check whether all weekends are empty
+        hide_weekend = (not any(map(itemgetter(1), events_by_day[5::7])) and
+                        not any(map(itemgetter(1), events_by_day[6::7])))
+        if hide_weekend:
+            events_by_day = filter(lambda x: x[0].weekday() not in (5, 6), events_by_day)
+
+        return WPCategory.render_template(template, self.category, events_by_day=events_by_day,
+                                          hide_weekend=hide_weekend, **params)
+
+    def _get_week_days(self):
+        # Return the days shown in the weekly overview
+        return self._get_days(self.start_dt, self.end_dt)
+
+    def _get_calendar_days(self):
+        # Return the days shown in the monthly overview
+        start_dt = self.start_dt - relativedelta(days=self.start_dt.weekday())
+        end_dt = self.end_dt + relativedelta(days=(7 - self.end_dt.weekday()) % 7)
+        return self._get_days(start_dt, end_dt)
+
+    @staticmethod
+    def _get_days(start_dt, end_dt):
+        # Return all days in the open-ended interval
+        current_dt = start_dt
+        while current_dt < end_dt:
+            yield current_dt
+            current_dt = current_dt + relativedelta(days=1)
+
+    @staticmethod
+    def _pop_head_while(predicate, l):
+        # Pop the head of the list while the predicate is true and return the popped elements
+        res = []
+        while len(l) and predicate(l[0]):
+            res.append(l[0])
+            l.pop(0)
+        return res
+
+    def _other_day_url(self, date):
+        return url_for('.overview', self.category, detail=self.detail, period=self.period,
+                       date=format_date(date, 'YYYY-MM-dd'))
+
+    def _process_multiday_events(self, event):
+        # Add "fake" proxy events for events spanning multiple days such that there is one event per day
+        # Function type: Event -> List[Event]
+        tzinfo = self.category.display_tzinfo
+
+        class _EventProxy(object):
+            def __init__(self, event, date):
+                start_dt = datetime.combine(date, event.start_dt.astimezone(tzinfo).timetz())
+                assert date >= event.start_dt
+                assert date <= event.end_dt
+                object.__setattr__(self, '_start_dt', start_dt)
+                object.__setattr__(self, '_real_event', event)
+
+            def __getattribute__(self, name):
```

Comment:
Why not use `__getattr__`? That way attributes that are set on the object itself are handled normally and only undefined attributes use your getter function